### PR TITLE
py-torchmetrics: New variant: image, deprecate @1.3.0

### PR DESCRIPTION
**Pull request by qwertos**
Depends on: #46257 and #46256

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-torchmetrics/package.py [hunk 1]
```diff
@@ -62,5 +68,11 @@ class PyTorchmetrics(PythonPackage):
         depends_on("py-lightning-utilities@0.7:", when="@1:")
         depends_on("py-pretty-errors@1.2.25", when="@1.4:")
 
+        depends_on("py-scipy@1:", when="+image")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-torchmetrics/package.py]:
```suggestion
        depends_on("py-scipy@1.0.1:", when="+image")
```

File: var/spack/repos/builtin/packages/py-torchmetrics/package.py [hunk 2]
```diff
@@ -62,5 +68,11 @@ class PyTorchmetrics(PythonPackage):
...
+        depends_on("py-torchvision@0.8:", when="+image")
+        depends_on("py-torch-fidelity@:0.3.0", when="@0.11.2:1.0.3+image")
+        depends_on("py-torch-fidelity@:0.4.0", when="@1.1:+image")
+        depends_on("py-lpips@:0.1.4", when="@:1.2.0+image")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-torchmetrics/package.py]: The upper bounds shouldn't be necessary, and get removed when generating the wheel anyway.

**Review by adamjstewart — COMMENTED** [review]
Out of curiosity, why do you want the variant?

**Comment by qwertos** [comment]
It's being requested by dalle2-pytorch. I'll push that up soon.

**Review by adamjstewart — APPROVED** [review]

**Review by bernhardkaindl — APPROVED** [review]
Build results on Ubuntu 22.04.4 LTS:
```py
bin/spack find --variants py-scipy@1.14.1%gcc@13 py-scipy@1.14.1%gcc@12 py-scipy@1.14.1%gcc@11 py-scipy@1.14.1%gcc@10
-- linux-ubuntu22.04-skylake / gcc@10.5.0 -----------------------
py-scipy@1.14.1

-- linux-ubuntu22.04-skylake / gcc@11.4.0 -----------------------
py-scipy@1.14.1

-- linux-ubuntu22.04-skylake / gcc@12.3.0 -----------------------
py-scipy@1.14.1

-- linux-ubuntu22.04-skylake / gcc@13.1.0 -----------------------
py-scipy@1.14.1
```
Generated by https://github.com/spack/build-quality-tools/blob/main/build_pr_changes.py
```py
/home/bkaindl/build-quality-tools/build_pr_changes.py -b py-scipy@1.14.1 -c gcc@13,gcc@12,gcc@11,gcc@10 -may
```

**Review by bernhardkaindl — APPROVED** [review]
Build results on Ubuntu 22.04.4 LTS:
```py
bin/spack find --variants py-torchmetrics+image%gcc@13 py-torchmetrics+image%gcc@12 py-torchmetrics+image%gcc@11 py-torchmetrics+image%gcc@10
-- linux-ubuntu22.04-skylake / gcc@10.5.0 -----------------------
py-torchmetrics@1.4.2+image

-- linux-ubuntu22.04-skylake / gcc@11.4.0 -----------------------
py-torchmetrics@1.4.2+image

-- linux-ubuntu22.04-skylake / gcc@12.3.0 -----------------------
py-torchmetrics@1.4.2+image

-- linux-ubuntu22.04-skylake / gcc@13.1.0 -----------------------
py-torchmetrics@1.4.2+image
```
Generated by https://github.com/spack/build-quality-tools/blob/main/build_pr_changes.py
```py
/home/bkaindl/build-quality-tools/build_pr_changes.py -b py-torchmetrics+image -c gcc@13,gcc@12,gcc@11,gcc@10 -may
```

**Review by bernhardkaindl — APPROVED** [review]
Build results on Ubuntu 22.04.4 LTS:
```py
bin/spack find --variants py-torchmetrics+image@1.3.0%clang@14.0.0 py-torchmetrics+image@1.3.0%gcc@9.5.0 py-torchmetrics+image@1.3.0%gcc@13.1.0 py-torchmetrics+image@1.3.0%gcc@12.3.0 py-torchmetrics+image@1.3.0%gcc@11.4.0 py-torchmetrics+image@1.3.0%gcc@10.5.0
-- linux-ubuntu22.04-skylake / clang@14.0.0 ---------------------
py-torchmetrics@1.3.0+image

-- linux-ubuntu22.04-skylake / gcc@9.5.0 ------------------------
py-torchmetrics@1.3.0+image

-- linux-ubuntu22.04-skylake / gcc@10.5.0 -----------------------
py-torchmetrics@1.3.0+image

-- linux-ubuntu22.04-skylake / gcc@11.4.0 -----------------------
py-torchmetrics@1.3.0+image

-- linux-ubuntu22.04-skylake / gcc@12.3.0 -----------------------
py-torchmetrics@1.3.0+image

-- linux-ubuntu22.04-skylake / gcc@13.1.0 -----------------------
py-torchmetrics@1.3.0+image
```
Generated by https://github.com/spack/build-quality-tools/blob/main/build_pr_changes.py
```py
/home/bkaindl/build-quality-tools/build_pr_changes.py -c all -may
```

**Pull request merged**